app/lib/methods/subscriptions/room.js

```diff
@@ -87,21 +89,36 @@ export default class RoomSubscription {
 		if (this.rid !== _rid) {
 			return;
 		}
-		if (ev === 'typing') {
+
+		const { UI_Use_Real_Name, Use_New_Action_Indicator } = reduxStore.getState().settings;
+		if (ev === 'user-activity' || (Use_New_Action_Indicator === false && ev === 'typing')) {
```

Comment:
No need to check the setting, you should always accept the old event for compatibility reasons since mobile and server are completely separated releases and users will get updates on mobile automatically meaning that they will have the new app but connected to old servers.